lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -194,6 +219,26 @@ public function avg($expression)
     }
 
     /**
+     * Adds a case statement for a branch of the $switch operator.
+     *
+     * Requires {@link switch()} to be called first. The argument can be any
+     * valid expression that resolves to a boolean. If the result is not a
+     * boolean, it is coerced to a boolean value.
+     *
+     * @param mixed|self $expression
+     *
+     * @return $this
+     */
+    protected function caseInternal($expression)
```

Comment:
No - since I consider it part of the public interface I thought it best to allow overriding it just like all other stage methods as well.